goalc/emitter/IGen.h

```diff
@@ -2155,14 +2162,10 @@ class IGen {
 
   // todo, rip relative loads and stores.
 
-  static Instruction mul_vf(Register dst, Register src1, Register src2) {
-    assert(dst.is_xmm());
-    assert(src1.is_xmm());
-    assert(src2.is_xmm());
-    Instruction instr(0x59);
-    instr.set_vex_modrm_and_rex(dst.hw_id(), src2.hw_id(), VEX3::LeadingBytes::P_0F, src1.hw_id());
-    return instr;
-  }
+  // TODO - rip, i reimplemented this below, which is better? im biased now
+  // personally, i think the shuffle instruction is a pain to work with directly, so id prefer
+  // working with it via helper functions (my splat one for example) so it doesn't really matter to
+  // me
```

Comment:
Keep both? I agree the helper is better in 99% of cases and yours is well designed, but it may be nice to have the "raw" version as well.